shelley-mc/formal-spec/utxo.tex

```diff
@@ -233,18 +233,18 @@ \subsection{UTxO Transitions}
       ~
       \\
       \mathsf{adaID}~\notin \dom~{\fun{forge}~tx} \\
-      \forall txout \in \txouts{txb}, ~ \fun{getValue}~txout  ~\geq ~ \epsilon \\~
+      \forall txout \in \txouts{txb}, ~ \fun{getValue}~txout  ~\geq ~ 0 \\~
+      \forall txout \in \txouts{txb}, ~ \fun{getCoin}~txout ~\geq \\
+      \fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue} pp \\~
```

Comment:
is the `valueSize` calculation defined anywhere? I am not sure if it needs to be in the formal spec or not. Maybe ask in the rule-based-spec channel? 
It should really be `scaledValueSize`, or equivalently
`(\fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue} pp ) / (size (the Coin type)`

The function `valueSize` should be something like
```(the number of PolicyIDs) * (size PolicyID) +
(the total number of AssetIDs) * (size (Quantity) + maxSize AssetID) ...```

Reply:
I named `valueSize` in analogy to `txSize`, and I didn't see an implementation for that at that time, but now I've seen that it's in an Appendix of the Shelley spec. Reading that, I've realized that there is also an issue with `txSize`, because we actually need a function like `valueSize` to compute `txSize`. I guess we should make an Appendix for that.

Scaling the size makes sense, otherwise the semantics of `minUTxOValue` changes after Shelley, so I'll do that.

Reply:
After noticing the issues with `txSize` in Shelley vs Goguen, I don't think we can scale it with the size of `Coin`, instead we have to use the size of a `Value` that only contains Ada.

Reply:
We should use the size of Coin, but scale the ada-only case of CompactValue by the scaling factor (size of Coin)